### PR TITLE
Product test refactor

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

During my work on #7485 I have realized that we have missed proper serialization groups on product variant. I have fixed that and also provide some small refactor of the test suite.

### CODE REVIEW

File: docs/api/product_variants.rst [hunk 1]
```diff
@@ -134,36 +134,13 @@ Example Response
...
                     "id": 1,
                     "code": "319bf720-e91d-36f5-aa8e-8c674e3861fb-variant-0",
-                    "option_values": [
-                        {
-                            "name": "Mug type",
-                            "code": "mug_type_medium"
-                        }
-                    ],
                     "position": 0,
-                    "translations": {
```
Comment (pjedrzejewski) [inline docs/api/product_variants.rst]: Why?

File: docs/api/product_variants.rst [hunk 2]
```diff
@@ -134,36 +134,13 @@ Example Response
                 {
                     "id": 1,
                     "code": "319bf720-e91d-36f5-aa8e-8c674e3861fb-variant-0",
-                    "option_values": [
```
Comment (pjedrzejewski) [inline docs/api/product_variants.rst]: I think this is very useful info, why removing it? :)

Comment (lchrusciel) [inline docs/api/product_variants.rst]: It is just the matter of product serialization. Right now, I just adjusted tests to the current implementation. It is not a problem to bring them back, but I will have to adjust the docs also.

**Comment by pjedrzejewski** [comment]
Thanks Łukasz!

**Pull request merged**